src/runtime/arch_tinygoriscv.go

```diff
@@ -90,3 +90,7 @@ func procPin() {
 func procUnpin() {
 	riscv.EnableInterrupts(procPinnedMask)
 }
+
+func waitForEvents() {
+	riscv.Asm("wfi")
```

Comment:
Reading the RISC-V specification, I think the same race condition is present. I'm not sure how to work around that, but there appears to be a trick with disabling/enabling interrupts.